add-on/chrome/bootstrap.js

```diff
@@ -1480,10 +1478,22 @@ function shutdown(data, reason) {
   // Stop waiting for browser windows to open.
   wm.removeListener(WindowListener);
 
-  // unregister about: handlers
-  AboutLoop.conversation.unregister(); // eslint-disable-line no-undef
-  AboutLoop.panel.unregister(); // eslint-disable-line no-undef
-  AboutLoop.toc.unregister(); // eslint-disable-line no-undef
+  // XXX do we really want the global PMM here, since it (unnecessarily, I
+  // think) sends a message to the CIPMM (is there any alternative?)
+  let ppmm = Cc["@mozilla.org/parentprocessmessagemanager;1"]
+           .getService(Ci.nsIMessageBroadcaster);
+  ppmm.broadcastAsyncMessage("LoopShuttingDown");
+
+  // XXX should we be listening for a response?  I'm not sure what we'd do if
+  // we got one, so it doesn't really seem worth it
+  AboutLoop.conversation.unregister();
+  AboutLoop.panel.unregister();
+  AboutLoop.toc.unregister();
+
+  // XXX this probably races with the broadcastAsyncMessage above, but it'd
+  // be a bit surprising if that actually matters
+  Services.ppmm.removeDelayedProcessScript(
```

Comment:
There's no race here - process scripts don't get removed on shutdown (same as for frame scripts).